packages/frontend/src/domain/sagas/authenticationSaga.ts

```diff
@@ -44,6 +60,44 @@ export function* signInWithEmailAndPasswordSaga(
   }
 }
 
+export function* signInWithGoogleSaga(): SagaIterator {
+  try {
+    const { idToken } = yield call(logInAsync, {
+      clientId: '',
+      iosClientId: GOOGLE_IOS_CLIENT_ID,
+      androidClientId: GOOGLE_ANDROID_CLIENT_ID,
+    });
+    yield put(setLoading(true));
+    const googleCredential = yield call(createGoogleCredential, idToken);
+    const User = yield call(signInWithCredential, googleCredential);
+    yield put(updateUser(User));
+  } catch (error) {
+    yield put(showAlert(error.message));
+  } finally {
+    yield put(setLoading(false));
+  }
+}
+
+export function* signInWithFacebookSaga(): SagaIterator {
+  try {
+    const { token } = yield call(
+      logInWithReadPermissionsAsync,
+      FACEBOOK_APP_ID,
+      {
+        permissions: ['public_profile', 'email'],
+      },
+    );
+    yield put(setLoading(true));
+    const facebookCredential = yield call(createFacebookCredential, token);
+    const User = yield call(signInWithCredential, facebookCredential);
```

Comment:
aqui eu estou tento problemas por que o signInWithCredential com a credencial do facebook nao esta retornando email dentro do user e sim dentro do providerData. To avaliando a opcao de so retirar o login pelo facebook.

Reply:
Olhando a doc deles:
https://firebase.google.com/docs/reference/js/firebase.auth.Auth.html#signinwithcredential
Retorna https://firebase.google.com/docs/reference/js/firebase.auth.html#usercredential
que tem https://firebase.google.com/docs/reference/js/firebase.User.html

email eh opcional, o que mostra que essa funcao ta errada
https://github.com/mtg-community/trading-card-manager/blob/35ccf6f3b38a23fc8e4c42967ec53ff3d284ecf0/packages/frontend/src/data/firebase/authentication.ts#L6-L9

Acho que basta acertar o mapper que vai dar bom